src/extensions/surveys.tsx

```diff
@@ -179,10 +179,8 @@ export class SurveyManager {
     }
 
     private clearSurveyTimeout(surveyId: string) {
-        if (this.surveyTimeouts.has(surveyId)) {
-            clearTimeout(this.surveyTimeouts.get(surveyId))
-            this.surveyTimeouts.delete(surveyId)
-        }
+        clearTimeout(this.surveyTimeouts.get(surveyId))
+        this.surveyTimeouts.delete(surveyId)
```

Comment:
you still do a dual lookup since delete has to check if it exists or not before deleting
```
const timeout = this.surveyTimeouts.get(surveyId)
if (time) {
    clearTimeout(timeout)
    this.surveyTimeouts.delete(surveyId)
}
```

now at least it only does the 2nd lookup if needed

Reply:
even then, it's a O(1) operation, so i don't think we gain much from this. i'll add though since no strong objections

Reply:
thats how systems get slow, one extra lookup here, another there, etc